internal/presentation/cli_test.go

```diff
@@ -0,0 +1,347 @@
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeT,
+								Treasure: 'A',
+							},
+							Rotation: model.Rotation0},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+┘.└
+.A.
+───
+`[1:], writer.String())
+		}
+
+		{
+			writer := new(strings.Builder)
+			drawer.DrawTo(writer, &model.Board{
+				Tiles: [][]*model.BoardTile{
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeT,
+								Treasure: 'B',
+							},
+							Rotation: model.Rotation90},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+│.└
+│B.
+│.┌
+`[1:], writer.String())
+		}
+
+		{
+			writer := new(strings.Builder)
+			drawer.DrawTo(writer, &model.Board{
+				Tiles: [][]*model.BoardTile{
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeT,
+								Treasure: 'C',
+							},
+							Rotation: model.Rotation180,
+						},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+───
+.C.
+┐.┌
+`[1:], writer.String())
+		}
+
+		{
+			writer := new(strings.Builder)
+			drawer.DrawTo(writer, &model.Board{
+				Tiles: [][]*model.BoardTile{
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeT,
+								Treasure: 'D',
+							},
+							Rotation: model.Rotation270},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+┘.│
+.D│
+┐.│
+`[1:], writer.String())
+		}
+
+		// V-shaped til
+		{
+			writer := new(strings.Builder)
+			drawer.DrawTo(writer, &model.Board{
+				Tiles: [][]*model.BoardTile{
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeV,
+								Treasure: 'A',
+							},
+							Rotation: model.Rotation0},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+──┐
+.A│
+┐.│
+`[1:], writer.String())
+		}
+
+		{
+			writer := new(strings.Builder)
+			drawer.DrawTo(writer, &model.Board{
+				Tiles: [][]*model.BoardTile{
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeV,
+								Treasure: 'B',
+							},
+							Rotation: model.Rotation90},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+┘.│
+.B│
+──┘
+`[1:], writer.String())
+		}
+
+		{
+			writer := new(strings.Builder)
+			drawer.DrawTo(writer, &model.Board{
+				Tiles: [][]*model.BoardTile{
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeV,
+								Treasure: model.NoTreasure,
+							},
+							Rotation: model.Rotation180},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+│.└
+│..
+└──
+`[1:], writer.String())
+		}
+
+		{
+			writer := new(strings.Builder)
+			drawer.DrawTo(writer, &model.Board{
+				Tiles: [][]*model.BoardTile{
+					{
+						{
+							Tile: &model.Tile{
+								Shape:    model.ShapeV,
+								Treasure: model.NoTreasure,
+							},
+							Rotation: model.Rotation270},
+					},
+				},
+			})
+
+			assert.Equal(t, `
+┌──
+│..
+│.┌
+`[1:], writer.String())
+		}
+	})
+
+	t.Run("Should draw a board", func(t *testing.T) {
+		drawer := &boardDrawer{}
+
+		writer := new(strings.Builder)
+		drawer.DrawTo(writer, &model.Board{
+			Tiles: [][]*model.BoardTile{
```

Comment:
I believe you could use a utility to create a board based on a string - that would make future tests easier to read. Of course, not this test.

Reply:
Good idea, I will check with to PO to see if it is out of scope for this use story or not.

Reply:
Saw with Florian, it is included in save user story scope.